src/voters/services/voterOutreach.service.ts

```diff
@@ -21,158 +21,122 @@ import { CrmCampaignsService } from '../../campaigns/services/crmCampaigns.servi
 import { EmailTemplateName } from 'src/email/email.types'
-import { OutreachService } from 'src/outreach/services/outreach.service'
-import { VoterFileType } from '../voterFile/voterFile.types'
+import { VoterFileFilterService } from './voterFileFilter.service'
+import { OutreachWithVoterFileFilter } from '../../outreach/types/outreach.types'
+
+export interface OutreachSlackBlocksConfiguration {
+  user: User
+  campaign: Campaign
+  type: OutreachType
+  date: Date
+  voterFileUrl: string
+  script: string
+  imageUrl?: string | null
+  message?: string
+  formattedAudience?: AudienceSlackBlock[]
+  audienceRequest: string
+}
+
+export type Audience = {
+  audience_superVoters?: boolean | null
+  audience_likelyVoters?: boolean | null
+  audience_unreliableVoters?: boolean | null
+  audience_unlikelyVoters?: boolean | null
+  audience_firstTimeVoters?: boolean | null
+  party_independent?: boolean | null
+  party_democrat?: boolean | null
+  party_republican?: boolean | null
+  age_18_25?: boolean | null
+  age_25_35?: boolean | null
+  age_35_50?: boolean | null
+  age_50_plus?: boolean | null
+  gender_male?: boolean | null
+  gender_female?: boolean | null
+}
 
 @Injectable()
 export class VoterOutreachService {
-  private readonly logger = new Logger(VoterOutreachService.name)
-
   constructor(
     private readonly slack: SlackService,
-    private readonly filesService: FilesService,
     private readonly campaignsService: CampaignsService,
     private readonly crmCampaigns: CrmCampaignsService,
-    private readonly emailService: EmailService,
-    private readonly textCampaignService: OutreachService,
+    private readonly email: EmailService,
+    private readonly voterFileFilterService: VoterFileFilterService,
   ) {}
 
+  private formatAudienceFiltersForSlack(
+    audience: Partial<Audience>,
+  ): Array<AudienceSlackBlock> {
+    return Object.entries(audience)
+      .filter(([key]) => key !== 'audience_request')
+      .map(([key, value]) => ({
+        type: SlackMessageType.RICH_TEXT_SECTION,
+        elements: [
+          {
+            type: SlackMessageType.TEXT,
+            text: `${key}: `,
+            style: {
+              bold: true,
+            },
+          },
+          {
+            type: SlackMessageType.TEXT,
+            text: value ? '✅ Yes' : '❌ No',
+          },
+        ],
+      }))
+  }
+
+  private buildVoterFileUrl({
+    audience,
+    type,
+    campaignSlug,
+  }: {
+    audience: Audience
+    type: OutreachType
+    campaignSlug: string
+  }): string {
+    const audienceFilters = Object.entries(audience).reduce(
+      (acc, [k, v]) => (v ? [...acc, k] : acc),
+      [] as string[],
+    )
+
+    const encodedFilters = audienceFilters
+      ? encodeURIComponent(JSON.stringify({ filters: audienceFilters }))
+      : null
+    return `${WEBAPP_ROOT}${WEBAPP_API_PATH}${VOTER_FILE_ROUTE}?type=${type}&slug=${campaignSlug}&customFilters=${encodedFilters}`
```

Comment:
Assumes `/` are all in the right place for these path segments. Path util here?

Reply:
These are app environment variables that we should for-sure validate.  But should prolly be done in the app bootstrap kicked off in /src/maint.ts, not here.

It's untouched from the original code, so it shouldn't cause any problems to leave as is: https://github.com/thegoodparty/gp-api/pull/308/files/71475bbcc2756b85cba8d9bd4762eeee58763e46#diff-11ea3d31ee418dd08dd31f469441a1bae02e80b08652c9cf4386222f74aa5c33L75